frontend/javascripts/dashboard/dataset/import_delete_component.js

```diff
@@ -0,0 +1,61 @@
+// @flow
+
+import { Button } from "antd";
+import React, { useState, useEffect } from "react";
+import * as Utils from "libs/utils";
+
+import type { APIDataset, APIDatasetId } from "admin/api_flow_types";
+import { getDataset, deleteDatasetOnDisk } from "admin/admin_rest_api";
+import Toast from "libs/toast";
+import messages from "messages";
+
+import { confirmAsync } from "./helper_components";
+
+type Props = {
+  datasetId: APIDatasetId,
+};
+
+export default function ImportSharingComponent({ datasetId }: Props) {
+  const [isDeleting, setIsDeleting] = useState(false);
+  const [dataSet, setDataSet] = useState<?APIDataset>(null);
+
+  async function fetch() {
+    const newDataSet = await getDataset(datasetId);
+    setDataSet(newDataSet);
+  }
+
+  useEffect(() => {
+    fetch();
+  }, []);
+
+  async function handleDeleteButtonClicked(): Promise<void> {
+    if (!dataSet) {
+      return;
+    }
+    await confirmAsync({
+      title: "Deleting a dataset on disk cannot be undone. Are you certain?",
+      okText: "Yes, Delete Dataset on Disk now",
+    });
+    setIsDeleting(true);
+    await deleteDatasetOnDisk(dataSet.dataStore.url, datasetId);
+    Toast.success(
+      messages["dataset.delete_success"]({
+        datasetName: dataSet.name,
+      }),
+    );
+    await Utils.sleep(2000);
+    setIsDeleting(false);
+    location.href = "/dashboard";
```

Comment:
We usually use the `RouterHistory` to redirect manually:
here is an [example](https://github.com/scalableminds/webknossos/blob/4fe8507278a037281a10c396fe33ab0dbce33753/frontend/javascripts/admin/auth/login_view.js#L22)

Could you please use it here too?